-1

**Failed Tests**:   Build  HeaderConsistency 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b47c2/15074/summary.html
**COMMIT**: dba6b5a2cf0d3ec772dedae51f8bbaccdfd077e9
**CMSSW**: CMSSW_12_0_X_2021-05-12-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33715/15074/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>Copying tmp/slc7_amd64_gcc900/src/GeneratorInterface/Hydjet2Interface/src/GeneratorInterfaceHydjet2Interface/libGeneratorInterfaceHydjet2Interface.so to productstore area:
Leaving library rule at GeneratorInterface/Hydjet2Interface
Entering library rule at src/GeneratorInterface/Hydjet2Interface/test
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-12-2300/src/GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-12-2300/src/GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc: In member function 'virtual void Hydjet2Analyzer::analyze(const edm::Event&, const edm::EventSetup&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-12-2300/src/GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc:554:23: error: 'Cos' is not a member of 'TMath'
  554 |           v2 = TMath::Cos(2 * (phiTrue));
      |                       ^~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-12-2300/src/GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc:555:23: error: 'Cos' is not a member of 'TMath'
  555 |           v3 = TMath::Cos(3 * (phiTrue - phi3));
      |                       ^~~
</pre>

--- a/FastSimulation/TrackingRecHitProducer/plugins/FastTrackerRecHitMatcher.cc
+++ b/FastSimulation/TrackingRecHitProducer/plugins/FastTrackerRecHitMatcher.cc
@@ -1,4 +1,5 @@
 // system include files
+#include <cfloat>
 #include <memory>
 
 // framework stuff
@@ -214,8 +215,8 @@ std::unique_ptr<FastTrackerRecHit> FastTrackerRecHitMatcher::match(const FastSin
   // position of the initial and final point of the strip in local coordinates (mono det)
   StripPosition stripmono = StripPosition(topol.localPosition(RPHIpointini), topol.localPosition(RPHIpointend));
 
-  if (trackdirection.mag2() <
-      FLT_MIN) {  // in case of no track hypothesis assume a track from the origin through the center of the strip
+  // in case of no track hypothesis assume a track from the origin through the center of the strip
+  if (trackdirection.mag2() < FLT_MIN) {
     LocalPoint lcenterofstrip = monoRH->localPosition();
     GlobalPoint gcenterofstrip = (stripdet->surface()).toGlobal(lcenterofstrip);
     GlobalVector gtrackdirection = gcenterofstrip - GlobalPoint(0, 0, 0);

--- a/GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc
+++ b/GeneratorInterface/Hydjet2Interface/test/Hydjet2Analyzer.cc
@@ -14,6 +14,7 @@
 //
 //
 // system include files
+#include <cmath>
 #include <memory>
 #include <iostream>
 #include <string>
@@ -551,25 +552,25 @@ void Hydjet2Analyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &i
             phiTrue = phi;
           }
           ///
-          v2 = TMath::Cos(2 * (phiTrue));
-          v3 = TMath::Cos(3 * (phiTrue - phi3));
-          v4 = TMath::Cos(4 * (phiTrue));
-          v5 = TMath::Cos(5 * (phiTrue - phi3));
-          v6 = TMath::Cos(6 * (phiTrue));
+          v2 = std::cos(2 * (phiTrue));
+          v3 = std::cos(3 * (phiTrue - phi3));
+          v4 = std::cos(4 * (phiTrue));
+          v5 = std::cos(5 * (phiTrue - phi3));
+          v6 = std::cos(6 * (phiTrue));
 
           if (doHistos_) {
             //common histos
             if ((*it)->status() < 10) {  //status 1
               dheta->Fill(eta);
 
-              if (TMath::Abs(eta) < 0.8) {
+              if (std::abs(eta) < 0.8) {
                 dhpt->Fill(pt);
                 dhphi->Fill(phiTrue);
               }
 
-              if (TMath::Abs(pdg_id) == 211 || TMath::Abs(pdg_id) == 321 || TMath::Abs(pdg_id) == 2212) {  //ch
+              if (std::abs(pdg_id) == 211 || std::abs(pdg_id) == 321 || std::abs(pdg_id) == 2212) {  //ch
 
-                if (TMath::Abs(eta) < 0.8) {
+                if (std::abs(eta) < 0.8) {
                   dhv0pt->Fill(pt, 1.);
                   dhv2pt->Fill(pt, v2);
                   dhpt_ch->Fill(pt);
@@ -587,12 +588,11 @@ void Hydjet2Analyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &i
                                 (((*it)->status() > 10) && (uStatus_ == 2)))) {  //user status
 
               //set1
-              if (TMath::Abs(pdg_id) == uPDG_1 || TMath::Abs(pdg_id) == uPDG_2 ||
-                  TMath::Abs(pdg_id) == uPDG_3) {  //uPDG
+              if (std::abs(pdg_id) == uPDG_1 || std::abs(pdg_id) == uPDG_2 || std::abs(pdg_id) == uPDG_3) {  //uPDG
                 if ((uStatus_ == 3) && ((*it)->status() < 10))
                   cout << "ustatus=3, but stab. part. found!!!" << endl;
 
-                if (TMath::Abs(eta) > downTetaCut_ && TMath::Abs(eta) < upTetaCut_) {  //eta cut
+                if (std::abs(eta) > downTetaCut_ && std::abs(eta) < upTetaCut_) {  //eta cut
 
                   uhv0pt->Fill(pt, 1.);
                   uhv2pt->Fill(pt, v2);

--- a/GeneratorInterface/HydjetInterface/test/HydjetAnalyzer.cc
+++ b/GeneratorInterface/HydjetInterface/test/HydjetAnalyzer.cc
@@ -6,6 +6,7 @@
 */
 
 // system include files
+#include <cmath>
 #include <memory>
 #include <iostream>
 #include <string>
@@ -577,25 +578,25 @@ void HydjetAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &iS
             phiTrue = phi;
           }
           ///
-          v2 = TMath::Cos(2 * (phiTrue));
-          v3 = TMath::Cos(3 * (phiTrue - phi3));
-          v4 = TMath::Cos(4 * (phiTrue));
-          v5 = TMath::Cos(5 * (phiTrue - phi3));
-          v6 = TMath::Cos(6 * (phiTrue));
+          v2 = std::cos(2 * (phiTrue));
+          v3 = std::cos(3 * (phiTrue - phi3));
+          v4 = std::cos(4 * (phiTrue));
+          v5 = std::cos(5 * (phiTrue - phi3));
+          v6 = std::cos(6 * (phiTrue));
 
           if (doHistos_) {
             //common histos
             if ((*it)->status() < 10) {  //status 1
               dheta->Fill(eta);
 
-              if (TMath::Abs(eta) < 0.8) {
+              if (std::abs(eta) < 0.8) {
                 dhpt->Fill(pt);
                 dhphi->Fill(phiTrue);
               }
 
-              if (TMath::Abs(pdg_id) == 211 || TMath::Abs(pdg_id) == 321 || TMath::Abs(pdg_id) == 2212) {  //ch
+              if (std::abs(pdg_id) == 211 || std::abs(pdg_id) == 321 || std::abs(pdg_id) == 2212) {  //ch
 
-                if (TMath::Abs(eta) < 0.8) {
+                if (std::abs(eta) < 0.8) {
                   dhv0pt->Fill(pt, 1.);
                   dhv2pt->Fill(pt, v2);
                   dhpt_ch->Fill(pt);
@@ -613,12 +614,11 @@ void HydjetAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &iS
                                 (((*it)->status() > 10) && (uStatus_ == 2)))) {  //user status
 
               //set1
-              if (TMath::Abs(pdg_id) == uPDG_1 || TMath::Abs(pdg_id) == uPDG_2 ||
-                  TMath::Abs(pdg_id) == uPDG_3) {  //uPDG
+              if (std::abs(pdg_id) == uPDG_1 || std::abs(pdg_id) == uPDG_2 || std::abs(pdg_id) == uPDG_3) {  //uPDG
                 if ((uStatus_ == 3) && ((*it)->status() < 10))
                   cout << "ustatus=3, but stab. part. found!!!" << endl;
 
-                if (TMath::Abs(eta) > downTetaCut_ && TMath::Abs(eta) < upTetaCut_) {  //eta cut
+                if (std::abs(eta) > downTetaCut_ && std::abs(eta) < upTetaCut_) {  //eta cut
 
                   uhv0pt->Fill(pt, 1.);
                   uhv2pt->Fill(pt, v2);

--- a/SimCalorimetry/HGCalAssociatorProducers/plugins/MultiClusterAssociatorByEnergyScoreImpl.cc
+++ b/SimCalorimetry/HGCalAssociatorProducers/plugins/MultiClusterAssociatorByEnergyScoreImpl.cc
@@ -8,6 +8,8 @@
 #include "SimDataFormats/CaloAnalysis/interface/SimCluster.h"
 #include "SimCalorimetry/HGCalAssociatorProducers/interface/AssociatorTools.h"
 
+#include <cfloat>
+
 MultiClusterAssociatorByEnergyScoreImpl::MultiClusterAssociatorByEnergyScoreImpl(
     edm::EDProductGetter const& productGetter,
     bool hardScatterOnly,

--- a/SimCalorimetry/HcalSimAlgos/interface/HcalHitFilter.h
+++ b/SimCalorimetry/HcalSimAlgos/interface/HcalHitFilter.h
@@ -5,6 +5,9 @@
 #include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
 #include "DataFormats/HcalDetId/interface/HcalDetId.h"
 
+#include <algorithm>
+#include <vector>
+
 template <HcalSubdetector... subdets>
 class HcalHitFilter : public CaloVHitFilter {
 public:

